ui/src/app/applications/components/application-details/application-resource-list.tsx

```diff
@@ -6,88 +6,120 @@ import {ResourceIcon} from '../resource-icon';
 import {ResourceLabel} from '../resource-label';
 import {ComparisonStatusIcon, HealthStatusIcon, nodeKey} from '../utils';
 import {Consumer} from '../../../shared/context';
+import {services} from '../../../shared/services';
 
 export const ApplicationResourceList = ({
     resources,
     onNodeClick,
-    nodeMenu
+    nodeMenu,
+    application
 }: {
-    resources: models.ResourceStatus[];
+    resources: any[];
     onNodeClick?: (fullName: string) => any;
     nodeMenu?: (node: models.ResourceNode) => React.ReactNode;
-}) => (
-    <div className='argo-table-list argo-table-list--clickable'>
-        <div className='argo-table-list__head'>
-            <div className='row'>
-                <div className='columns small-1 xxxlarge-1' />
-                <div className='columns small-2 xxxlarge-2'>NAME</div>
-                <div className='columns small-3 xxxlarge-4'>GROUP/KIND</div>
-                <div className='columns small-3 xxxlarge-3'>NAMESPACE</div>
-                <div className='columns small-2 xxxlarge-2'>STATUS</div>
+    application?: models.Application;
+}) => {
+    const [resourceRows, setResourceRows] = React.useState(resources);
+    React.useEffect(() => {
+        setResourceRows(resources);
+        async function getLiveState() {
+            if (application) {
+                const liveStatePromises = resources.map(async resource => {
+                    const resourceRow: any = {...resource};
+                    const liveState = await services.applications.getResource(application.metadata.name, resource).catch(() => null);
+                    if (liveState?.metadata?.annotations && liveState?.metadata?.annotations[models.AnnotationHookKey]) {
+                        resourceRow.syncOrder = liveState?.metadata.annotations[models.AnnotationHookKey];
+
+                        if (liveState?.metadata?.annotations[models.AnnotationSyncWaveKey]) {
+                            resourceRow.syncOrder = resourceRow.syncOrder + ': ' + liveState?.metadata.annotations[models.AnnotationSyncWaveKey];
+                        }
+                    }
+                    return resourceRow;
+                });
+                const resourcesWithSyncOrder = await Promise.all(liveStatePromises);
+                if (resourcesWithSyncOrder.length > 0) {
+                    setResourceRows(resourcesWithSyncOrder);
+                }
+            }
+        }
+        getLiveState();
+    }, [resources]);
+    return (
+        <div className='argo-table-list argo-table-list--clickable'>
+            <div className='argo-table-list__head'>
+                <div className='row'>
+                    <div className='columns small-1 xxxlarge-1' />
+                    <div className='columns small-2 xxxlarge-2'>NAME</div>
+                    <div className='columns small-2 xxxlarge-2'>GROUP/KIND</div>
+                    <div className='columns small-1 xxxlarge-2'>SYNC ORDER</div>
```

Comment:
Would it be possible to hide this entire column if there's no order found in the resources?

Reply:
Good catch! Problem resolved. Thanks :)